jazz_create-serverless-service/test/test.js

```diff
@@ -22,234 +22,667 @@ const awsContext = require('aws-lambda-mock-context');
+        checkCase("principalId", null, undefined, errMessage, errType);
       assert.isTrue(bothCases);
     });
-
     /*
-    * Given an event with an invalid rateExpression, handler() indicates the expression is invalid
-    * @param {object} event, contains a body.rateExpression that is empty, null, or invalid
-    * @params {object, function} default aws context and callback function as assigned above respectively
-    * @returns index.handler() should return a descriptive InternalServer error notification
-    */
-    it("should inform the user the rateExpression is invalid if given a faulty rateExpression", ()=>{
-      var cronValues = [null, "", "P!nk!e_P!e"];
-      var errMessage1 = "Empty Cron expression.";
-      var errMessage2 = "Invalid Cron expression. ";
-      var errType = "InternalServerError";
-      var allCases = checkCase("body", "rateExpression", cronValues[0], errMessage1, errType) &&
-                      checkCase("body", "rateExpression", cronValues[1], errMessage1, errType) &&
-                      checkCase("body", "rateExpression", cronValues[2], errMessage2, errType);
-      assert.isTrue(allCases);
+     * Given successful parameters and setup, handler() should send a POST http request
+     * @params {object, object, function} default event, aws context, callback
+     * @returns index.handler() should attempt an http POST if given valid paramters
+     */
+    it("should give success message if service onboarding in Jenkins setup attempt is succesfull", () => {
+      let responseObject_getToken = {
+        statusCode: 200,
+        body: {
+          data: {
+            "token": "ghd93-3240-2343"
+          }
+        }
+      };
+      let responseObject_createService = {
+        statusCode: 200,
+        body: {
+          data: {
+            "service_id": "ghd93-3240-2343"
+          }
+        }
+      };
+      let responseObject_serviceOnboarding = {
+        statusCode: 200,
+        body: {
+          message: "Service Creation Success"
+        }
+      };
+      event.stage = "dev";
+      let config = configObj(event);
+      // wrapping requests 
+      reqStub = sinon.stub(request, "Request", (obj) => {
+        // Matching response Object to the corresponding Request call
+        if (obj.uri === (config.SERVICE_API_URL + config.TOKEN_URL)) {
+          return obj.callback(null, responseObject_getToken, responseObject_getToken.body);
+        } else if (obj.uri === "https://{conf-apikey}.execute-api.{conf-region}.amazonaws.com/dev/jazz/services") {
+
+          return obj.callback(null, responseObject_createService, responseObject_createService.body);
+
+        } else if (obj.url === '{conf-jenkins-host}/job/create-service/buildWithParameters') {
+
+          return obj.callback(null, responseObject_serviceOnboarding, responseObject_serviceOnboarding.body);
+        }
+        //return obj.callback(null, responseObject, responseObject.body);
```

Comment:
remove this and other commented code in other places

Reply:
done